libpod/image/image.go

```diff
@@ -1075,3 +1077,62 @@ func (i *Image) Comment(ctx context.Context, manifestType string) (string, error
 	}
 	return ociv1Img.History[0].Comment, nil
 }
+
+// Tree gets dependencies of image
+func (i *Image) Tree(ctx context.Context) (*bytes.Buffer, error) {
+	treeDepth := 1
+	rootImage := i
+
+	//Traverse to the root(base) image
+	parent, err := i.GetParent()
+	if err != nil {
+		return nil, err
+	}
+
+	for parent != nil {
+		rootImage = parent
+		parent, err = parent.GetParent()
+		if err != nil {
+			break
+		}
+		treeDepth++
+	}
+
+	var buffer bytes.Buffer
+	//Print all subtree(peers) to the depth of input image
+	err = printSubTree(&buffer, rootImage, 0, "└─ ", treeDepth)
+
+	return &buffer, err
+}
+
+func printSubTree(buffer *bytes.Buffer, root *Image, rootSize uint64, prefix string, treeDepth int) error {
+	if treeDepth == 0 {
+		return nil
+	}
+	size, err := root.Size(context.Background())
+	if err != nil {
+		return err
+	}
+	//Size of current layer will be current layer size minus upper layer
+	buffer.WriteString(fmt.Sprintf("%s ID: %s  Tags: %v VirtualSize: %s\n",
+		prefix,
+		root.ID()[:12],
+		root.Names(),
+		units.HumanSizeWithPrecision(float64(*size-rootSize), 4)),
+	)
+
+	children, err := root.GetChildren()
+	if err != nil {
+		return nil
+	}
+
+	for _, child := range children {
+		img, err := root.imageruntime.getImage(child)
+		if err != nil {
+			break
+		}
+		nextPrefix := "  "
```

Comment:
(Non-blocking: This assigns a constant in every loop iteration, with only one use; this could just as well be inlined as `"  "+prefix` one line below, and it could be a tiny bit more readable.  The spaces are not a “next prefix”, and naming the variable correctly is much harder than not having the variable :) )